### PR TITLE
Added additional core engine

### PR DESCRIPTION
**Pull request by tractorcow**
Rebased from https://github.com/silverstripe-australia/silverstripe-queuedjobs/pull/47

### CODE REVIEW

**Comment by tractorcow** [comment]
@nyeholt I've squashed and included the refactor of the top level core engine. Please let me know if you like this one. :D

File: docs/en/configuring-multi-process-execution.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	Injector:
+	  ProcessJobQueueTask:
+	    properties:
+	      TaskRunner: %$DoormanRunner
+
+
+By default, this will allow a single child process to complete queued jobs. You can increase the number of processes allowed by changing the default rule:
+
+
+:::yaml
```
Comment (assertchris) [inline docs/en/configuring-multi-process-execution.md]: `\t`

**Comment by nyeholt** [comment]
Am just playing around with a couple refactorings that might be worthwhile - give me an hour or so  :)

**Comment by tractorcow** [comment]
I've added another commit to this based on yours (with a few tweaks). Hope this suits. :D

**Comment by nyeholt** [comment]
Ah yes, I missed that runJob bit. The circular dependency is a bit of an issue - I kind of lean towards pushing the service through as a parameter rather than a singleton, as the singleton call  is essentially the same as the wired circular dependency.

**Comment by tractorcow** [comment]
Hi @nyeholt  just checking in.  Any further action you'd like from me on this PR? :)

**Comment by nyeholt** [comment]
Oh! I Thought you'd merged it already :) I think I'll change the getService to pushing the service down as a parameter, but I don't have the time to do so right now so instead of holding things up I'll merge now and change later.

**Comment by nyeholt** [comment]
Have tagged a new minor version (2.8.0) too.

**Pull request merged**